src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs

```diff
@@ -50,8 +64,11 @@ public static IServiceCollection Configure<TOptions>(this IServiceCollection ser
                 throw new ArgumentNullException(nameof(configureOptions));
             }
 
-            services.AddSingleton<IConfigureOptions<TOptions>>(new ConfigureOptions<TOptions>(configureOptions));
+            services.AddSingleton<IConfigureOptions<TOptions>>(new ConfigureNamedOptions<TOptions>(name, configureOptions));
             return services;
         }
+
+        public static IServiceCollection ConfigureAll<TOptions>(this IServiceCollection services, Action<TOptions> configureOptions) where TOptions : class
+            => services.Configure(name: null, configureOptions: configureOptions);
```

Comment:
Let's imagine the following:

```
services.Configure<MyOptions>(config1);
services.ConfigureAll<MyOptions>(config2);
services.Configure<MyOptions>("name", config3);
```

According the current realization `OptionsFactory` we will get:

```
var options1 = Get<MyOptions>(); // will be applied: config1 and config2
var options2 = Get<MyOptions>("name"); // will be applied: config1, config2 and config3
```

If I'm not missing something, then `ConfigureAll` does and works the same as unnamed `Configure`.

Or you're seeing another case?